test/tests.js

```diff
@@ -593,11 +640,94 @@ describe( 'passport-saml /', function() {
       };
     }
 
+    function testForCheckLogout( check ) {
+      return function( done ) {
+        var app = express();
+        app.use( bodyParser.urlencoded({extended: false}) );
+        app.use( passport.initialize() );
+        var config = check.config;
+        config.callbackUrl = 'http://localhost:3033/login';
+        config.entryPoint = 'https://wwwexampleIdp.com/saml';
+        var profile = null;
+        passport.use( new SamlStrategy( config, function(_profile, done) {
+            profile = _profile;
+            done(null, profile );
+          })
+        );
+
+        var userSerialized = false;
+        passport.serializeUser(function(user, done) {
+          userSerialized = true;
+          done(null, user);
+        });
+
+        app.post( '/login',
+          passport.authenticate( "saml"),
+          function(req, res) {
+            res.status(200).send("200 OK");
+          });
+
+        app.use( function( err, req, res, next ) {
+          // console.log( err.stack );
+          res.status(500).send('500 Internal Server Error');
+        });
+
+        server = app.listen( 3033, function() {
+          var requestOpts = {
+            url: 'http://localhost:3033/login',
+            method: 'post',
+            form: check.samlRequest
+          };
+
+          request(requestOpts, function(err, response, body) {
+            try {
+              var encodedSamlResponse = querystring.parse(this.uri.query).SAMLResponse;
+              // An error will exist because the endpoint we're trying to log out of doesn't exist,
+              // but we can still test to make sure that everything is behaving as it should.
+              // should.not.exist(err);
+
+              var buffer = Buffer.from(encodedSamlResponse, 'base64');
+              if (check.config.skipRequestCompression)
+                helper(null, buffer);
+              else
+                zlib.inflateRaw( buffer, helper );
+
+              function helper(err, samlResponse) {
+                try {
+                  should.not.exist( err );
+                  parseString( samlResponse.toString(), function( err, doc ) {
+                    try {
+                      should.not.exist( err );
+                      delete doc['samlp:LogoutResponse']['$']["ID"];
+                      delete doc['samlp:LogoutResponse']['$']["IssueInstant"];
+                      doc.should.eql( check.result );
+                      done();
+                    } catch (err2) {
+                      done(err2);
+                    }
+                  });
+                } catch (err2) {
+                  done(err2);
+                }
+              }
+            } catch (err2) {
```

Comment:
Uuh, try-catch hell =/. More descriptive error names might help. But since this is coming from upstream, we shouldn't probably change this here.

Reply:
Agreed, but as you said we don'tt want to differ from upstream as it would complicate future merges. :/